python/tests/test_pyvw.py

```diff
@@ -20,6 +20,7 @@ def test_learn_predict(self):
         assert init == 0
         self.model.learn(ex)
         assert self.model.predict(ex) > init
+        assert self.model.predict({'x': "| a:1 b:0.5"}) == 0
```

Comment:
When using a dictionary to create an example it uses the value as the feature name, the parser is not run. Therefore this value does not make sense and is confusing to someone reading the tests

Reply:
To reword Jack, this line should actually fail. So keep this test but revert the assert.

Reply:
So should it be changed to this `assert self.model.predict({"| a:1 b:0.5":'a'}) == 0`?